index.js

```diff
@@ -2787,6 +2787,28 @@
   }
   S.prop = def('prop', {a: [Accessible]}, [$.String, a, b], prop);
 
+  //# props :: Accessible a => Array String -> a -> b
+  //.
+  //. Takes an array of property names and an object with known properties and
+  //. returns the value of the specified property. If for some reason the
+  //. object lacks the specified property, a type error is thrown.
+  //.
+  //. For accessing properties of uncertain objects, use [`gets`](#gets)
+  //. instead.
+  //.
+  //. ```javascript
+  //. > S.props(['a', 'b', 'c'], {a: {b: {c: 1}}})
+  //. 1
+  //. ```
+  function props(path, obj) {
+    return path.reduce(function(memo, key) {
+      if (key in memo) return memo[key];
+      throw new TypeError('‘props’ expected object to have a property at ‘' +
+        R.toString(path) + '’; ' + R.toString(obj) + ' does not');
```

Comment:
I don't see a reason to quote the path; `[` and `]` will show clearly where it begins and ends.

Also, let's tweak the error message (and alignment) like so:

```javascript
      throw new TypeError('‘props’ expected object to have a property path ' +
                          R.toString(path) + '; ' +
                          R.toString(obj) + ' does not');
```